### PR TITLE
Normalize addresses in name controller

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## Explanation

Normalize addresses and chain IDs in the `NameController` state by storing them in lowercase, to avoid persisting multiple names for the same address. The UI components in the extension won't display the lowercase address but instead will render the checksummed address.

Also replace the `getChainId` callback with the `variation` option in the `SetNameRequest` and `UpdateProposedNamesRequest`. This decouples the controller from the selected network and supports requests on any chain.

### `@metamask/name-controller`

- **BREAKING**: Save addresses and chain IDs as lowercase in state.
- **BREAKING**: Remove `getChainId` constructor callback.
- **BREAKING**: Require a `variation` property when calling `setName` or `updateProposedNames` with the address type.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/name-controller/src/NameController.ts [hunk 1]
```diff
@@ -151,10 +148,10 @@ export class NameController extends BaseControllerV2<
   setName(request: SetNameRequest) {
     this.#validateSetNameRequest(request);
 
-    const { value, type, name, sourceId: requestSourceId } = request;
```
Comment (OGPoyraz) [inline packages/name-controller/src/NameController.ts:154]: Should we add variation to the jsdoc?

File: packages/name-controller/src/NameController.ts [hunk 2]
```diff
@@ -174,12 +171,10 @@ export class NameController extends BaseControllerV2<
   ): Promise<UpdateProposedNamesResult> {
     this.#validateUpdateProposedNamesRequest(request);
 
-    const chainId = this.#getChainId();
-
     const providerResponses = (
       await Promise.all(
         this.#providers.map((provider) =>
-          this.#getProviderResponse(request, chainId, provider),
+          this.#getProviderResponse(request, provider),
```
Comment (OGPoyraz) [inline packages/name-controller/src/NameController.ts:177]: `variation` missing on this jsdoc as well

File: packages/name-controller/src/NameController.ts [hunk 3]
```diff
@@ -151,10 +148,10 @@ export class NameController extends BaseControllerV2<
   setName(request: SetNameRequest) {
     this.#validateSetNameRequest(request);
 
-    const { value, type, name, sourceId: requestSourceId } = request;
+    const { value, type, name, sourceId: requestSourceId, variation } = request;
```
Comment (OGPoyraz) [inline packages/name-controller/src/NameController.ts:151]: Should we add `variation` to the jsdoc?

File: packages/name-controller/src/NameController.ts (hunk 2 above)
Comment (matthewwalsh0) [reply packages/name-controller/src/NameController.ts:179]: Done.

File: packages/name-controller/src/NameController.ts (hunk 3 above)
Comment (matthewwalsh0) [reply packages/name-controller/src/NameController.ts:152]: Done.

**Review by OGPoyraz — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Pull request merged**